adapters/coap-vertx-base/src/main/java/org/eclipse/hono/adapter/coap/DeviceInfoSupplier.java

```diff
@@ -0,0 +1,135 @@
+/**
+ * Copyright (c) 2021 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+
+package org.eclipse.hono.adapter.coap;
+
+import java.security.Principal;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.concurrent.CopyOnWriteArrayList;
+
+import org.eclipse.californium.elements.auth.AdditionalInfo;
+import org.eclipse.californium.scandium.auth.AdvancedApplicationLevelInfoSupplier;
+import org.eclipse.californium.scandium.auth.ApplicationLevelInfoSupplier;
+import org.eclipse.hono.auth.Device;
+import org.eclipse.hono.tracing.TracingHelper;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import io.opentracing.Span;
+import io.opentracing.Tracer;
+import io.opentracing.tag.Tags;
+
+/**
+ * Device information supplier.
+ */
+public class DeviceInfoSupplier implements ApplicationLevelInfoSupplier, AdvancedApplicationLevelInfoSupplier {
+
+    /**
+     * Key in the extended info of the peer identity containing the auth-id used for authentication.
+     */
+    public static final String EXT_INFO_KEY_HONO_AUTH_ID = "hono-auth-id";
+    /**
+     * Key in the extended info of the peer identity containing the authenticated Device.
+     */
+    public static final String EXT_INFO_KEY_HONO_DEVICE = "hono-device";
+
+    private static final Logger LOG = LoggerFactory.getLogger(DeviceInfoSupplier.class);
+
+    /**
+     * The vert.x context to run interactions with Hono services on.
+     */
```

Comment:
this comment doesn't seem to match here

Reply:
That class may change much more, when we know, if resumption must be supported and if a non-blocking API will be used. 

My favorite is then just

```
    public AdditionalInfo getInfo(final Principal clientIdentity, final Object customArgument) {
        if (customArgument instanceof AdditionalInfo) {
            return (AdditionalInfo) customArgument;
        }
        LOG.warn("missing custom argument of type AdditionalInfo");
        return AdditionalInfo.empty();
    }
```

Which should obsolete the Tracer and adapter name.

Reply:
Comment removed.